tls/s2n_handshake_io.c

```diff
@@ -694,7 +694,16 @@ static int s2n_advance_message(struct s2n_connection *conn)
 }
 
 int s2n_generate_new_client_session_id(struct s2n_connection *conn)
+    CONTRACT_REQUIRES(__CPROVER_is_fresh(conn, sizeof(*conn)) || (conn == NULL))
+    CONTRACT_ASSIGNS_ERR(conn->session_id_len, conn->session_id)
+    CONTRACT_ENSURES(S2N_IMPLIES(conn == NULL, __CPROVER_return_value == S2N_FAILURE))
+    CONTRACT_ENSURES(S2N_IMPLIES(__CPROVER_return_value == S2N_SUCCESS && conn->mode == S2N_SERVER, conn->session_id_len == S2N_TLS_SESSION_ID_MAX_LEN))
```

Comment:
What about "__CPROVER_return_value == S2N_SUCCESS && conn->mode == S2N_CLIENT"?

Reply:
So, in this case the routine does nothing as far as I can see, and since we say that the only things change are conn->session_id and conn->session_id_len, what we would want to say is just that these two values don't change. Unfortunately there is a limitation in the tool right now that prevents us saying this. The two commented out lines do say that, but causes the tool to run off into the weeds. I will file an issue on that.

Should I perhaps file an issue here to add those clauses once that issue is fixed?